samples/create-symbol-styles-from-web-styles/src/main/java/com/esri/arcgismaps/sample/createsymbolstylesfromwebstyles/components/CreateSymbolStylesFromWebStylesViewModel.kt

```diff
@@ -0,0 +1,184 @@
+    // portal parameter
+    private val symbolStyle = SymbolStyle.createWithStyleNameAndPortal("Esri2DPointSymbolsStyle", null)
+
+    // Create a list of the required symbol names in the web style
+    private val symbolNames = listOf(
+        "atm",
+        "beach",
+        "campground",
+        "city-hall",
+        "hospital",
+        "library",
+        "park",
+        "place-of-worship",
+        "police-station",
+        "post-office",
+        "school",
+        "trail"
+    )
+
+    // Create a flow to hold the symbol names and icons
+    private val _symbolNamesAndIconsFlow = MutableStateFlow<List<Pair<String, BitmapDrawable>>>(emptyList())
+    val symbolNamesAndIconsFlow = _symbolNamesAndIconsFlow.asStateFlow()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Load the symbols into a list
+        viewModelScope.launch {
+            _symbolNamesAndIconsFlow.value = createUniqueValueSymbols()
+        }
+
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map", error.message.toString()
+                )
+            }
+        }
+    }
+
+    /**
+     * Create a series of unique values and add them to the unique value renderer. Also create a list of the symbol
+     * names and icons and return them as a list of pairs.
+     */
+    private suspend fun createUniqueValueSymbols(): MutableList<Pair<String, BitmapDrawable>> {
+        // Create a list to hold the symbol names and icons
+        val symbolNamesAndIconsList = mutableListOf<Pair<String, BitmapDrawable>>()
+        symbolNames.forEach { symbolName ->
+            // Search for each symbol in the symbol style
+            symbolStyle.getSymbol(listOf(symbolName)).onSuccess { symbol ->
+                // Get a list of all categories to be mapped to the symbol
+                val categories = mapSymbolNameToField(symbolName)
+                categories.forEach { category ->
+                    // Add each unique value category to the unique value renderer
+                    uniqueValueRenderer.uniqueValues.add(
+                        UniqueValue(
+                            label = symbolName, symbol = symbol, values = listOf(category)
+                        )
+                    )
+                }
+                // Create a swatch from the symbol
+                symbol.createSwatch(
+                    screenScale = 2.0f, width = 30.0f, height = 30.0f, backgroundColor = Color.white
+                ).onSuccess { symbolBitmap ->
+                    // Add the symbol name and symbol to the list
+                    symbolNamesAndIconsList.add(Pair(symbolName, symbolBitmap))
+                }
+            }
+        }
```

Comment:
Should we handle `onFailure` here? Or, does that feel unnecessary for the sample? 🤔